test/webapp-strategy-test.js

```diff
@@ -538,41 +541,68 @@ describe("/lib/strategies/webapp-strategy", function () {
 				done();
 			};
 			webAppStrategy.success = function (user) {
-				
+
 				assert.equal(options.successRedirect, "redirectUri");
 				assert.isObject(req.session[WebAppStrategy.AUTH_CONTEXT]);
-				
+
 				assert.isString(req.session[WebAppStrategy.AUTH_CONTEXT].accessToken);
 				assert.equal(req.session[WebAppStrategy.AUTH_CONTEXT].accessToken, "access_token_mock");
 				assert.isObject(req.session[WebAppStrategy.AUTH_CONTEXT].accessTokenPayload);
 				assert.equal(req.session[WebAppStrategy.AUTH_CONTEXT].accessTokenPayload.scope, "appid_default");
-				
+
 				assert.isString(req.session[WebAppStrategy.AUTH_CONTEXT].identityToken);
 				assert.equal(req.session[WebAppStrategy.AUTH_CONTEXT].identityToken, "id_token_mock");
 				assert.isObject(req.session[WebAppStrategy.AUTH_CONTEXT].identityTokenPayload);
 				assert.equal(req.session[WebAppStrategy.AUTH_CONTEXT].identityTokenPayload.scope, "appid_default");
-				
+
 				assert.isObject(user);
 				assert.equal(user.scope, "appid_default");
-				
+
 				done();
 			};
-			
+
 			var req = {
 				session: {},
 				query: {
 					code: "WORKING_CODE",
 					state: ""
 				}
 			};
-			
+
 			var options = {
 				successRedirect: "redirectUri"
 			};
 			req.session[WebAppStrategy.STATE_PARAMETER] = { anonymousLogin : false , state : "123456789" };
 			webAppStrategy.authenticate(req, options);
 		});
-		
+
+		it("Should fail if the verify callback is called with an error", function (done) {
+			webAppStrategy.fail = function (err) {
+				assert.equal(err, "Missing ID token");
+				done();
+			};
+			webAppStrategy.success = function () {
+				assert.equal(req.session[WebAppStrategy.AUTH_CONTEXT].identityTokenPayload.scope, "appid_default");
```

Comment:
This should always fail so if we get to success we want to throw an error. We can accomplish this by making the only thing in this function be something like `done('Should fail')`